src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -494,14 +501,20 @@ fn all_constructors<'a, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
         }
         ty::TyUint(uint_ty) if exhaustive_integer_patterns => {
             use syntax::ast::UintTy::*;
-            let (min, (max, ty)) = (0u128, match uint_ty {
-                Usize => (usize::MAX as u128, cx.tcx.types.usize),
-                U8    => (   u8::MAX as u128, cx.tcx.types.u8),
-                U16   => (  u16::MAX as u128, cx.tcx.types.u16),
-                U32   => (  u32::MAX as u128, cx.tcx.types.u32),
-                U64   => (  u64::MAX as u128, cx.tcx.types.u64),
-                U128  => ( u128::MAX as u128, cx.tcx.types.u128),
-            });
+            let min_max_ty = |sty| {
+                let size = cx.tcx.layout_of(ty::ParamEnv::reveal_all().and(sty))
```

Comment:
You should use `cx.param_env` or something. Also, you don't need to match on `uint_ty` at all! Pass `pcx.ty` instead.

Reply:
Oh, that's so much nicer, of course!